pkg/build/builder/docker_test.go

```diff
@@ -319,6 +320,63 @@ func TestEmptySource(t *testing.T) {
 		}
 	}
 }
+
+// We should not be able to try to push to scratch
+func TestDockerfileFromScratch(t *testing.T) {
+	strPointer := new(string)
+	*strPointer = `FROM scratch
+USER 1001`
+
+	dockerClient := &FakeDocker{
+		buildImageFunc: func(opts docker.BuildImageOptions) error {
+			return nil
+		},
+		pushImageFunc: func(opts docker.PushImageOptions, auth docker.AuthConfiguration) error {
+			if opts.Name == "scratch" && opts.Registry == "" {
+				return fmt.Errorf("cannot push scratch")
+			}
+			return nil
+		},
+	}
+
+	build := &api.Build{
+		Spec: api.BuildSpec{
+			CommonSpec: api.CommonSpec{
+				Source: api.BuildSource{
+					ContextDir: "",
+					Dockerfile: strPointer,
+				},
+				Strategy: api.BuildStrategy{
+					DockerStrategy: &api.DockerBuildStrategy{
+						DockerfilePath: "",
+						From: &kapi.ObjectReference{
+							Kind: "DockerImage",
+							Name: "scratch",
+						},
+					},
+				},
+				Output: api.BuildOutput{
+					To: &kapi.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: "scratch",
+					},
+				},
+			},
+		},
+	}
+
+	dockerBuilder := &DockerBuilder{
+		build:        build,
+		dockerClient: dockerClient,
+		gitClient:    git.NewRepository(),
+		tar:          tar.New(s2iutil.NewFileSystem()),
+	}
+
+	if err := dockerBuilder.Build(); err != nil {
+		t.Error("Should have not received error on docker build")
+	}
```

Comment:
need a test that actually confirms we get the error when you try to push to scratch.